src/cascade/input_data/emr.py

```diff
@@ -0,0 +1,219 @@
+        prevalence.loc[zero_means, "mean"] = not_quite_zero
+
+    MATHLOG.debug(f"{len(prevalence)} prevalence observations avaliable, exluding sex_id 4")
+    return prevalence
+
+
+def _collapse_times(csmr):
+    MATHLOG.debug("Treating CSMR measurements over time ranges as measurements at the ranges midpoint")
+    csmr["time"] = (csmr["time_lower"] + csmr["time_upper"]) / 2
+    return csmr.drop(columns=["time_lower", "time_upper"])
+
+
+def _collapse_ages_unweighted(csmr):
+    csmr["age"] = (csmr["age_lower"] + csmr["age_upper"]) / 2
+    return csmr
+
+
+def _collapse_ages_weighted(execution_context, csmr):
+    mean_years = get_mean_years(execution_context)
+    mean_years = mean_years.rename(columns={"mean": "age", "location_id": "node_id", "year_id": "time_lower"})
+    csmr = csmr.merge(mean_years, on=["age_group_id", "node_id", "sex_id", "time_lower"])
+    csmr["age"] += csmr["age_lower"]
+    return csmr
+
+
+def _prepare_csmr(execution_context, csmr):
+    MATHLOG.debug("Preparing CSMR data from GBD")
+    csmr = csmr.rename(columns={"location_id": "node_id"})
+
+    MATHLOG.debug("Assigning standard error from measured upper and lower.")
+    csmr = meas_bounds_to_stdev(csmr)
+
+    null_means = csmr["mean"].isnull()
+    if np.any(null_means):
+        MATHLOG.warn(
+            f"{np.sum(null_means)} CSMR rows with NaN means in EMR calculation. "
+            "Removing them but there may be something wrong"
+        )
+        csmr = csmr.loc[~null_means]
+
+    csmr = age_groups_to_ranges(execution_context, csmr, keep_age_group_id=True)
+    if execution_context.parameters.use_weighted_age_group_midpoints:
+        MATHLOG.debug("Treating CSMR measurements over age ranges as measurements at the mortality weighted midpoint")
+        csmr = _collapse_ages_weighted(execution_context, csmr)
+    else:
+        MATHLOG.debug("Treating CSMR measurements over age ranges as measurements at the unweighted midpoint")
+        csmr = _collapse_ages_unweighted(csmr)
+    csmr = csmr.drop(columns=["age_lower", "age_upper"])
+
+    csmr = _collapse_times(csmr)
+
+    return csmr
+
+
+def _make_interpolators(csmr):
+    """ Calculate interpolations over CSMR so we can integrate the CSMR value
+    over the age/time range of each prevalence measurement.
+    """
+    mean = {}
+    stderr = {}
+    mean["both"] = LSQBivariateSpline(
+        csmr.age, csmr.time, csmr["mean"], sorted(csmr.age.unique()), sorted(csmr.time.unique()), kx=1, ky=1
+    )
+    stderr["both"] = LSQBivariateSpline(
+        csmr.age, csmr.time, csmr.standard_error, sorted(csmr.age.unique()), sorted(csmr.time.unique()), kx=1, ky=1
+    )
+
+    # csmr_by_age = csmr.sort_values("age").groupby("age").mean().reset_index()
+    # mean["age"] = InterpolatedUnivariateSpline(csmr_by_age.age, csmr_by_age["mean"], k=1)
+    # stderr["age"] = InterpolatedUnivariateSpline(csmr_by_age.age, csmr_by_age.standard_error, k=1)
```

Comment:
I think a certain someone would ask me to remove these lines, if I left them in?

Reply:
Well, if it were me who were that certain someone, I would ask you to test your code better because that's not just a bug but a stupid one.